central/hash/manager/manager.go

```diff
@@ -0,0 +1,113 @@
+package manager
+
+import (
+	"context"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stackrox/rox/central/hash/datastore"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/sync"
+)
+
+const (
+	flushInterval = 1 * time.Minute
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+// Manager is a hash manager that provides access to cluster-based dedupers and persists
+// the hashes into the database
+type Manager interface {
+	Start(ctx context.Context)
+
+	GetDeduper(ctx context.Context, clusterID string) Deduper
+	Delete(ctx context.Context, clusterID string) error
+}
+
+// NewManager instantiates a Manager
+func NewManager(datastore datastore.Datastore) Manager {
+	return &managerImpl{
+		datastore: datastore,
+		dedupers:  make(map[string]Deduper),
+	}
+}
+
+type managerImpl struct {
+	datastore datastore.Datastore
+
+	dedupersLock sync.RWMutex
+	dedupers     map[string]Deduper
+}
+
+func (m *managerImpl) flushHashes(ctx context.Context) {
+	var hashesToFlush []*storage.Hash
+	concurrency.WithLock(&m.dedupersLock, func() {
+		hashesToFlush = make([]*storage.Hash, 0, len(m.dedupers))
```

Comment:
nitpick: Just a thought. Instead of cloning maps for all clusters and then flushing them all at once. Can we do it sequentially, cluster by cluster?

Some considerations that I had:
- smaller memory spike. We don't have to keep all duplicates in memory, just one at a time
- it could also reduce payload sent to DB in a single request